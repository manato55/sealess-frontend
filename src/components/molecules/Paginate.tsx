import React from 'react';
import { Pagination } from '@material-ui/lab';
import styled from 'styled-components';

interface Props {
  contents: object[];
  perPage: number;
  offset: number;
  change: (_: React.ChangeEvent<unknown>, page: number) => Promise<void>;
}

export const Paginate = (props: Props): React.ReactElement => {
  return (
    <>
      <PaginateWrapper>
        <div>
          <Pagination
            count={Math.ceil(props.contents?.length / props.perPage)}
            variant="outlined"
            shape="rounded"
            color="primary"
            page={props.offset}
            onChange={props.change}
            size="small"
            id="paginate"
          />
        </div>
      </PaginateWrapper>
    </>
  );
};

const PaginateWrapper = styled.div`
  text-align: center;
  margin-top: 50px;
  width: 90%;
  position: relative;

  div {
    position: absolute;
    right: 50%;
  }
`;

export default Paginate;
