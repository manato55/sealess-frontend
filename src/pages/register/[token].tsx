import RegisterToken from '../../components/organisms/RegisterToken';

const Token = () => {
  return (
    <>
      <RegisterToken />
    </>
  );
};

export default Token;
