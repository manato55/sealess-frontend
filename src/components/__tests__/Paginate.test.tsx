import React from 'react';
import Paginate from '../molecules/Paginate';
import { mount, shallow } from 'enzyme';
import 'jsdom-global/register';

describe('Paginate test', () => {
  let container;
  const onChangeMock = jest.fn();

  beforeEach(() => {
    container = shallow(<Paginate contents={[]} perPage={3} offset={3} change={onChangeMock} />);
  });

  test('onChange test', () => {
    const input = container.find('#paginate');
    input.simulate('change');
    expect(onChangeMock).toHaveBeenCalled();
  });

  test('snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
