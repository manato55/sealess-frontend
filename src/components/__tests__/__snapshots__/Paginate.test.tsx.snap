// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginate test snapshot 1`] = `
<Fragment>
  <styled.div>
    <WithStyles(ForwardRef(Pagination))
      color="primary"
      count={0}
      id="paginate"
      onChange={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      page={3}
      shape="rounded"
      size="small"
      variant="outlined"
    />
  </styled.div>
</Fragment>
`;
